posthog/clickhouse/test/test_person_overrides.py

```diff
@@ -85,3 +88,57 @@ def test_can_insert_person_overrides():
 
         sync_execute(DROP_KAFKA_PERSON_OVERRIDES_TABLE_SQL)
         sync_execute(DROP_PERSON_OVERRIDES_CREATE_MATERIALIZED_VIEW_SQL)
+
+
+class PersonOverrideValues(TypedDict):
+    """A dict of values that may be inserted into person_overrides."""
+
+    team_id: int
+    old_person_id: UUID
+    override_person_id: UUID
+    merged_at: datetime
+    oldest_event: datetime
+    created_at: datetime
+    version: int
+
+
+@pytest.mark.django_db
+def test_person_overrides_dict():
+    """Test behavior of person_overrides_dict with multiple versions of same key.
+
+    The dictionary should always favor the latest version after every reload.
+    """
+    sync_execute(PERSON_OVERRIDES_CREATE_TABLE_SQL)
+    sync_execute(PERSON_OVERRIDES_CREATE_DICTIONARY_SQL)
+
+    values: PersonOverrideValues = {
+        "team_id": 1,
+        "old_person_id": uuid4(),
+        "override_person_id": uuid4(),
+        "merged_at": datetime.fromisoformat("2020-01-02T00:00:00+00:00"),
+        "oldest_event": datetime.fromisoformat("2020-01-01T00:00:00+00:00"),
+        "created_at": datetime.utcnow(),
```

Comment:
Not really relevant but this value will be set automatically by ClickHouse fwiw, so you could exclude it from the insert.